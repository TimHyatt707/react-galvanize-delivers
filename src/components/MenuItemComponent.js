import React from 'react';

export default function MenuItemComponent({ item, onAddItem }) {
  function handleClick(event) {
    event.preventDefault();
    onAddItem(item);
  }
  return (
    <div className="MenuItemComponent col s12 m6">
      <div className="card">
        <div className="card-image">
          <img src={item.imagePath || 'N/a'} alt="N/a" />
        </div>
        <div className="card-content">
          <h5 className="card-title">
            {item.name || 'N/A'}
          </h5>
          <p>
            {item.price ? '$' + item.price.toFixed(2) : 'N/A'}
          </p>
        </div>
        <div className="card-action">
          <a onClick={handleClick}>ADD TO ORDER</a>
        </div>
      </div>
    </div>
  );
}
