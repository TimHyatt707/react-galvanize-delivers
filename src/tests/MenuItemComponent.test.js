import React from 'react';
// import { expect } from 'chai';
import { shallow, mount } from 'enzyme';

import MenuItemComponent from './../components/MenuItemComponent';

const item = {
  name: 'Some Item A',
  price: 9.99,
  imagePath: '//via.placeholder.com/300x200'
};

function handleClick(event) {
  event.preventDefault();
  onAddItem(item.id);
}

const wrapper = shallow(<MenuItemComponent item={item} />);

describe('Should render MenuItemComponent, name, price, and onAddItem', () => {
  const fullWrapper = mount(<MenuItemComponent item={item} />);
  const onAddItem = jest.fn();
  it('renders div component', () => {
    expect(wrapper.find('div')).toHaveLength(5);
  });
  it('renders item name', () => {
    expect(fullWrapper.props().item.name).toBe(item.name);
  });
  it('renders item price', () => {
    expect(fullWrapper.props().item.price).toBe(item.price);
  });
  it('should test if onAddItem function was fired', () => {
    const onAddItem = jest.fn();

    mount(<MenuItemComponent onAddItem={onAddItem} item={item} />)
      .find('a')
      .simulate('click');
    expect(onAddItem).toBeCalled();
  });
});
