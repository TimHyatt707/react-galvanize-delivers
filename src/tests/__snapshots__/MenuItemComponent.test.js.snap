// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MenuItemComponent, name, price, and onAddItem snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="MenuItemComponent col s12 m6"
  >
    <div
      className="card"
    >
      <div
        className="card-image"
      >
        <img
          alt="N/a"
          src="//via.placeholder.com/300x200"
        />
      </div>
      <div
        className="card-content"
      >
        <h5
          className="card-title"
        >
          Some Item A
        </h5>
        <p>
          $9.99
        </p>
      </div>
      <div
        className="card-action"
      >
        <a
          onClick={[Function]}
        >
          ADD TO ORDER
        </a>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="MenuItemComponent col s12 m6"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="N/a"
            src="//via.placeholder.com/300x200"
          />
        </div>
        <div
          className="card-content"
        >
          <h5
            className="card-title"
          >
            Some Item A
          </h5>
          <p>
            $9.99
          </p>
        </div>
        <div
          className="card-action"
        >
          <a
            onClick={[Function]}
          >
            ADD TO ORDER
          </a>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <MenuItemComponent
        item={
          Object {
            "imagePath": "//via.placeholder.com/300x200",
            "name": "Some Item A",
            "price": 9.99,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "imagePath": "//via.placeholder.com/300x200",
            "name": "Some Item A",
            "price": 9.99,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="MenuItemComponent col s12 m6"
        >
          <div
            className="card"
          >
            <div
              className="card-image"
            >
              <img
                alt="N/a"
                src="//via.placeholder.com/300x200"
              />
            </div>
            <div
              className="card-content"
            >
              <h5
                className="card-title"
              >
                Some Item A
              </h5>
              <p>
                $9.99
              </p>
            </div>
            <div
              className="card-action"
            >
              <a
                onClick={[Function]}
              >
                ADD TO ORDER
              </a>
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="MenuItemComponent col s12 m6"
        >
          <div
            className="card"
          >
            <div
              className="card-image"
            >
              <img
                alt="N/a"
                src="//via.placeholder.com/300x200"
              />
            </div>
            <div
              className="card-content"
            >
              <h5
                className="card-title"
              >
                Some Item A
              </h5>
              <p>
                $9.99
              </p>
            </div>
            <div
              className="card-action"
            >
              <a
                onClick={[Function]}
              >
                ADD TO ORDER
              </a>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MenuItemComponent
    item={
      Object {
        "imagePath": "//via.placeholder.com/300x200",
        "name": "Some Item A",
        "price": 9.99,
      }
    }
  />,
}
`;
